app/src/main/java/org/mozilla/fenix/settings/HomeSettingsFragment.kt

```diff
@@ -97,11 +96,20 @@ class HomeSettingsFragment : PreferenceFragmentCompat() {
         override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
             try {
                 val context = preference.context
-                context.components.analytics.metrics.track(
-                    Event.CustomizeHomePreferenceToggled(
-                        preference.key,
+                val telemetryAllowMap = mapOf(
+                    context.getString(R.string.pref_key_show_top_sites) to "most_visited_sites",
```

Comment:
will this be simpler if we move it to `setupPreferences` and remove `CustomizeHomeMetricsUpdater`?